pkg/pipeline/pipeline.go

```diff
@@ -43,13 +45,26 @@ const (
 
 // Pipeline manager
 type Pipeline struct {
-	IsRunning      bool
-	pipelineStages []*PipelineEntry
-	configStages   []config.Stage
-	configParams   []config.StageParam
+	startNodes    []*node.Init
+	terminalNodes []*node.Terminal
+	IsRunning     bool
+	// TODO: this field is only used for test verification. We should rewrite the build process
+	// to be able to remove it from here
+	pipelineStages []*pipelineEntry
+}
+
+// builder stores the information that is only required during the build of the pipeline
+type builder struct {
```

Comment:
@mariomac do you want to move the` builder` and all the functions into a dedicated file ... this file is packed and it make sense to split